src/commit.h

```diff
@@ -36,4 +36,17 @@ struct git_commit {
 void git_commit__free(void *commit);
 int git_commit__parse(void *commit, git_odb_object *obj);
 
+typedef enum {
+	GIT_COMMIT_PARSE_TREE = (1 << 0),
+	GIT_COMMIT_PARSE_PARENTS = (1 << 1),
+	GIT_COMMIT_PARSE_AUTHOR = (1 << 2),
+	GIT_COMMIT_PARSE_COMMITTER = (1 << 3),
+	GIT_COMMIT_PARSE_MESSAGE_ENCODING = (1 << 4),
+	GIT_COMMIT_PARSE_RAW_HEADER = (1 << 5),
+	GIT_COMMIT_PARSE_RAW_MESSAGE = (1 << 6),
+	GIT_COMMIT_PARSE_ALL = 0xFFFF,
+} git_commit__parse_flags;
```

Comment:
What's the purpose of those flags? Is it an optimization or is it that you cannot parse some stuff because of certain conditions? If it's for the purpose of optimization, I'd like to avoid doing that. We currently don't have any notion of "partial" objects, and I bet this could theoretically lead to subtle bugs when passing around those partial objects.

Reply:
Idem, revwalk parse merge.